authors/apps/authentication/tests/test_password_reset.py

```diff
@@ -1,21 +1,156 @@
+        """
+        response = self.client.post(
+            reverse('authentication:password-reset'),
+            self.empty_email,
+            format='json'
+        )
+        returned_data = json.loads(response.content)
+        self.assertIn('This field may not be blank', str(returned_data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
-    def test_password_reset_email_non_existent(self):
-        pass
+    def test_empty_email_field(self):
+        """
+        tests that an empty email field is not sent as a request
+        """
+        response = self.client.post(
+            reverse('authentication:password-reset'),
+            self.blank_email,
+            format='json'
+        )
+        returned_data = json.loads(response.content)
+        self.assertIn('This field is required', str(returned_data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
-    def test_password_reset_email_missing_request(self):
-        pass
+    def test_valid_password(self):
+        """
+        test for request with valid password
+        """
+        user = User.objects.get(email=self.user.email)
+        token = user.get_reset_token(10)
+        response = self.client.put(
+            reverse('authentication:set-updated-password',
+            kwargs={'reset_token': token}),
+            self.password,
+            format='json'
+        )
+        output = json.loads(response.content)
+        self.assertIn('you may now log into your account with', str(output))
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
 
-    def test_password_reset_email_invalid_pattern(self):
-        pass
+    def test_blank_password(self):
+        """
+        test for request with blank request object
+        """
+        user = User.objects.get(email=self.user.email)
+        token = user.get_reset_token(10)
+        response = self.client.put(
+            reverse('authentication:set-updated-password',
+            kwargs={'reset_token': token}),
+            self.blank_password,
+            format='json'
+        )
+        output = json.loads(response.content)
+        self.assertIn('This field is required.', str(output))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+    def test_empty_password(self):
+        """
+        test for request with empty password value in JSON object
+        """
+        user = User.objects.get(email=self.user.email)
+        token = user.get_reset_token(10)
+        response = self.client.put(
+            reverse('authentication:set-updated-password',
+            kwargs={'reset_token': token}),
+            self.empty_password,
+            format='json'
+        )
+        output = json.loads(response.content)
+        self.assertIn('This field may not be blank', str(output))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+    def test_short_password(self):
```

Comment:
I've noticed that you've catered for short passwords. Passwords should also not be less than 8 characters and shouldn't be only alphanumeric. Did you check for this too (in the tests and reset password functionality)?

Reply:
I have added a test to check an alphanumeric password